include/regs.h

```diff
@@ -92,12 +135,77 @@ constexpr auto BL_INDEX = 0;
 
 #endif
 
+// The CpuTestFlags struct allows CPU test flags to be applied en masse. First
+// construct it with the set of flags you will be adjusting followed by
+// assigning the actual flag state to the verbosely-named members. For example:
+//
+// CpuTestFlags my_flags(FLAG_CF | FLAG_OF);
+//   my_flags.has_carry = (did the operation need to carry results?);
+//   my_flags.has_overflow = (did the operation overflow?);
+//
+// Then apply the flags to the CPU registers:
+//   cpu_regs.ApplyTestFlags(my_flags);
+//
+struct CpuTestFlags {
+	constexpr CpuTestFlags() = delete;
+	constexpr CpuTestFlags(const uint32_t _clear_mask);
```

Comment:
Minor: you could call this `clear_mask` in the declaration.

Reply:
Nice improvement to the API 👍
Fixed.